setup.py

```diff
@@ -54,6 +54,7 @@
         "graphs": ["networkx~=3.2.1", "matplotlib~=3.8.1"],
         "websurfer": ["beautifulsoup4", "markdownify", "pdfminer.six", "pathvalidate"],
         "redis": ["redis"],
+        "ipython": ["jupyter-client>=8.6.0", "ipykernel>=6.29.0"],
```

Comment:
We document optional dependencies in https://microsoft.github.io/autogen/docs/installation/Optional-Dependencies

Reply:
Done. Also removed docker from optional dependencies